File: src/Prueba.java
public class Prueba extends Publicacion {

  private Club club;

  private Boolean activa = true;


  private CategoriaFormativa categoria;


  private LocalDate fechaCreacion;

  // Constructors
  public Prueba() {
    this.fechaCreacion = LocalDate.now();
  }

  public Prueba(String titulo, String contenido, String imagenUrl, Club club, CategoriaFormativa categoria) {
    super(titulo, contenido, imagenUrl);
    this.club = club;
    this.categoria = categoria;
    this.fechaCreacion = LocalDate.now();
  }

  public boolean desactivar() {
    if (activa) {
      this.activa = false;
      return true;
    }
    return false;
  }

}

File: app/model/entities/Prueba.ts
import { Club } from "./Club";
import { Publicacion } from "./Publicacion";
import { CategoriaFormativa } from "./CategoriaFormativa";

export class Prueba extends Publicacion {
  club?: Club;
  activa: boolean = true;
  categoria?: CategoriaFormativa;
  fechaCreacion?: Date;

  constructor(titulo?: string, contenido?: string, imagenUrl?: string, club?: Club, categoria?: CategoriaFormativa) {
    super(titulo, contenido, imagenUrl);
    if (club && categoria) {
      this.club = club;
      this.categoria = categoria;
    }
    this.fechaCreacion = new Date();
  }

  desactivar(): boolean {
    if (this.activa) {
      this.activa = false;
      return true;
    }
    return false;
  }

  setClub(club: Club): void { this.club = club; }
  setActiva(activa: boolean): void { this.activa = activa; }
  setCategoria(categoria: CategoriaFormativa): void { this.categoria = categoria; }
  setFechaCreacion(fecha: Date): void { this.fechaCreacion = fecha; }

}
